ord_schema/updates.py

```diff
@@ -0,0 +1,128 @@
+    """Resolves compound identifiers to SMILES via the PubChem REST API."""
+    response = urllib.request.urlopen(
+        'https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/'
+        f'{value_type}/{value}/property/IsomericSMILES/txt')
+    return response.read().decode().strip()
+
+
+def resolve_names(message):
+    """Attempts to resolve compound NAME identifiers to SMILES.
+
+    When a NAME identifier is resolved, a SMILES identifier is added to the list
+    of identifiers for that compound. Note that this function moves on to the
+    next Compound after the first successful name resolution.
+
+    Args:
+        message: Reaction proto.
+
+    Returns:
+        Boolean whether `message` was modified.
+    """
+    modified = False
+    compounds = message_helpers.find_submessages(message, reaction_pb2.Compound)
+    for compound in compounds:
+        if any(identifier.type in _COMPOUND_STRUCTURAL_IDENTIFIERS
+               for identifier in compound.identifiers):
+            continue  # Compound already has a structural identifier.
+        for identifier in compound.identifiers:
+            if identifier.type == identifier.NAME:
+                try:
+                    smiles = _pubchem_resolve('name', identifier.value)
+                    new_identifier = compound.identifiers.add()
+                    new_identifier.type = new_identifier.SMILES
+                    new_identifier.value = smiles
+                    new_identifier.details = 'NAME resolved by PubChem'
+                    modified = True
+                    break
+                except urllib.error.HTTPError:
+                    pass
+    return modified
+
+
+def add_binary_identifiers(message):
+    """Adds RDKIT_BINARY identifiers for compounds with valid structures.
+
+    Note that the RDKIT_BINARY representations are mostly useful in the context
+    of searching the database. Accordingly, this function is not included in the
+    standard set of Reaction updates in update_reaction().
+
+    Args:
+        message: Reaction proto.
+
+    Returns:
+        Boolean whether `message` was modified.
+    """
+    modified = False
+    compounds = message_helpers.find_submessages(message, reaction_pb2.Compound)
+    for compound in compounds:
+        if any(identifier.type == identifier.RDKIT_BINARY
+               for identifier in message.identifiers):
+            continue
+        for identifier in compound.identifiers:
+            mol = None
+            if Chem and identifier.type == identifier.SMILES:
+                mol = Chem.MolFromSmiles(identifier.value)
+            elif identifier.type == identifier.INCHI:
+                mol = Chem.MolFromInchi(identifier.value)
+            elif identifier.type == identifier.MOLBLOCK:
+                mol = Chem.MolFromMolBlock(identifier.value)
+            if mol is not None:
+                compound.identifiers.add(
+                    bytes_value=mol.ToBinary(), type='RDKIT_BINARY')
```

Comment:
Should we add `details` saying what the source of the `RDKIT_BINARY` was? This might not be necessary but thought I'd ask anyway. It looks like here you might end up with 3 `RDKIT_BINARY`s if there are identifiers for `SMILES`, `INCHI`, _and_ `MOLBLOCK` for whatever reason.

Reply:
Yes, that's a good idea.

Reply:
Also I'll update this so it breaks after the first RDKIT_BINARY.